### PR TITLE
ENH: Fix doc link in HTML representation of estimators

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR fixes the interactive HTML representation of sklearnex estimators that don't have a scikit-learn analog to offer a clickable link to their docs, the same way scikit-learn estimators do, which is currently not working.

Before:
image

After:
image

Note: here I'm using the "latest" tag, as I wasn't sure how to get the link to the actual version of sklearnex that is used at runtime. Currently, this "latest" tag points to a 2023 release, so it won't work as-is until that issue is fixed.

---

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable

### CODE REVIEW

File: sklearnex/_utils.py [hunk 1]
```diff
@@ -113,3 +114,19 @@ def get_hyperparameters(self, op):
...
+# before inheriting from a scikit-learn estimator, otherwise will get overriden
+# by the estimator's original.
+class NonSKLearnAlgorithm(ABC):
+    @property
+    def _doc_link_module(self) -> str:
+        return "sklearnex"
+
+    @property
+    def _doc_link_template(self) -> str:
+        module_path = ".".join(self.__class__.__module__.split(".")[:-1])
```
Comment (icfaust) [inline sklearnex/_utils.py:130]:
```suggestion
        module_path, _ = ".".join(self.__class__.__module__.rsplit(".", 1))

```

File: sklearnex/_utils.py [hunk 2]
```diff
@@ -113,3 +114,19 @@ def get_hyperparameters(self, op):
         return estimator_class
 
     return wrap_class
+
+
+# This abstract class is meant to generate a clickable doc link for classses
+# in sklearnex that are not present in scikit-learn. It should be inherited
+# before inheriting from a scikit-learn estimator, otherwise will get overriden
+# by the estimator's original.
+class NonSKLearnAlgorithm(ABC):
```
Comment (icfaust) [inline sklearnex/_utils.py:123]:
```suggestion
class IntelEstimator(ABC):

```

**Review by icfaust — COMMENTED** [review]
Good PR, just some subtle stuff:
1) I think describing what something has rather than what it doesn't is better, also switch ```Algorithm``` to ```Estimator``` to follow sklearn convention. I'd prefer naming it after Intel in this case since the source is from Intel, though we could swap to oneDAL or something.
2) Just some small thing on using rsplit instead of split.

We can merge this today no problem.

File: sklearnex/_utils.py (hunk 2 above)
Comment (david-cortes-intel) [reply sklearnex/_utils.py]: Renamed.

**Review by icfaust — APPROVED** [review]

File: sklearnex/_utils.py [hunk 3]
```diff
@@ -113,3 +114,20 @@ def get_hyperparameters(self, op):
...
+# before inheriting from a scikit-learn estimator, otherwise will get overriden
+# by the estimator's original.
+class IntelEstimator(ABC):
+    @property
+    def _doc_link_module(self) -> str:
+        return "sklearnex"
+
+    @property
+    def _doc_link_template(self) -> str:
+        module_path, _ = ".".join(self.__class__.__module__.rsplit(".", 1))
```
Comment (icfaust) [inline sklearnex/_utils.py:130]:
```suggestion
        module_path, _ = self.__class__.__module__.rsplit(".", 1)

```
Sorry, made a mistake in my original suggestion. Totally my fault.
